java/client/src/main/java/glide/api/models/configuration/RequestRoutingConfiguration.java

```diff
@@ -13,31 +13,46 @@ public class RequestRoutingConfiguration {
      * Basic interface. Please use one of the following implementations:
      *
      * <ul>
-     *   <li>{@link SimpleRoute}
+     *   <li>{@link SimpleMultiNodeRoute}
+     *   <li>{@link SimpleSingleNodeRoute}
      *   <li>{@link SlotIdRoute}
      *   <li>{@link SlotKeyRoute}
+     *   <li>{@link ByAddressRoute}
      * </ul>
      */
-    public interface Route {
-        boolean isSingleNodeRoute();
+    public interface Route {}
+
+    /** A route that addresses single node. */
+    public interface SingleNodeRoute extends Route {}
+
+    /** A route that addresses multiple nodes. */
+    public interface MultiNodeRoute extends Route {}
+
+    @RequiredArgsConstructor
+    @Getter
+    public enum SimpleSingleNodeRoute implements SingleNodeRoute {
+        /** Route request to a random node. */
+        RANDOM(2);
+
+        private final int ordinal;
     }
 
-    public enum SimpleRoute implements Route {
+    @RequiredArgsConstructor
+    @Getter
+    public enum SimpleMultiNodeRoute implements MultiNodeRoute {
         /** Route request to all nodes. */
-        ALL_NODES,
+        ALL_NODES(0),
         /** Route request to all primary nodes. */
-        ALL_PRIMARIES,
-        /** Route request to a random node. */
-        RANDOM;
+        ALL_PRIMARIES(1);
 
-        @Override
-        public boolean isSingleNodeRoute() {
-            return this == RANDOM;
-        }
+        private final int ordinal;
```

Comment:
what is the ordinal for?

Reply:
Can it be moved to a parent (`Route`?) and maybe we need an ordinal for all types?

Reply:
It is needed for enum only to ensure that we set it correctly in protobuf
https://github.com/aws/glide-for-redis/blob/0a430cf35db910745fde4bacf08100ee6ae7b56a/glide-core/src/protobuf/redis_request.proto#L4-L8

Reply:
When it was a single enum I ensured that by the same order as in proto file.